src/makeCalculator.js

```diff
@@ -4,7 +4,35 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+    add(val) {
+      this.result += val;
+    },
+    subtract(val) {
+      this.result -= val;
+    },
+    multiply(val) {
+      this.result *= val;
+    },
+    divide(val) {
+      this.result /= val;
```

Comment:
The `divide` method should handle division by zero to avoid potential errors. Consider adding a check to ensure `val` is not zero before performing the division.